nifi-commons/nifi-vault-utils/src/test/java/org/apache/nifi/vault/hashicorp/TestHashiCorpVaultConfiguration.java

```diff
@@ -0,0 +1,195 @@
+        return writeVaultAuthProperties(properties);
+    }
+
+    public static void writeProperties(Map<String, String> props, File authProps) throws IOException {
+        Properties properties = new Properties();
+
+        for (Map.Entry<String, String> entry : props.entrySet()) {
+            properties.put(entry.getKey(), entry.getValue());
+        }
+        try (Writer writer = new FileWriter(authProps)) {
+            properties.store(writer, "Vault test authentication properties");
+        }
+    }
+
+    public void runTest() {
+        config = new HashiCorpVaultConfiguration(propertiesBuilder.build());
+
+        VaultEndpoint endpoint = config.vaultEndpoint();
+        Assert.assertEquals("localhost", endpoint.getHost());
+        Assert.assertEquals(8200, endpoint.getPort());
+        Assert.assertEquals("http", endpoint.getScheme());
+
+        ClientAuthentication clientAuthentication = config.clientAuthentication();
+        Assert.assertNotNull(clientAuthentication);
+    }
+
+    @Test
+    public void testBasicProperties() {
+        this.runTest();
+    }
+
+    @Test
+    public void testTlsProperties() throws IOException {
+        propertiesBuilder.setKeyStore(keystoreFile.toFile().getAbsolutePath());
+        propertiesBuilder.setKeyStorePassword(KEYSTORE_PASSWORD_VALUE);
+        propertiesBuilder.setKeyStoreType(KEYSTORE_TYPE_VALUE);
+        propertiesBuilder.setTrustStore(truststoreFile.toFile().getAbsolutePath());
+        propertiesBuilder.setTrustStorePassword(TRUSTSTORE_PASSWORD_VALUE);
+        propertiesBuilder.setTrustStoreType(TRUSTSTORE_TYPE_VALUE);
+        propertiesBuilder.setEnabledTlsProtocols(TLS_V_1_3_VALUE);
+        propertiesBuilder.setEnabledTlsCipherSuites(TEST_CIPHER_SUITE_VALUE);
+
+        this.runTest();
+
+        SslConfiguration sslConfiguration = config.sslConfiguration();
+        Assert.assertEquals(keystoreFile.toFile().getAbsolutePath(), sslConfiguration.getKeyStoreConfiguration().getResource().getFile().getAbsolutePath());
+        Assert.assertEquals(KEYSTORE_PASSWORD_VALUE, new String(sslConfiguration.getKeyStoreConfiguration().getStorePassword()));
+        Assert.assertEquals(KEYSTORE_TYPE_VALUE, sslConfiguration.getKeyStoreConfiguration().getStoreType());
+        Assert.assertEquals(truststoreFile.toFile().getAbsolutePath(), sslConfiguration.getTrustStoreConfiguration().getResource().getFile().getAbsolutePath());
+        Assert.assertEquals(TRUSTSTORE_PASSWORD_VALUE, new String(sslConfiguration.getTrustStoreConfiguration().getStorePassword()));
+        Assert.assertEquals(TRUSTSTORE_TYPE_VALUE, sslConfiguration.getTrustStoreConfiguration().getStoreType());
+        Assert.assertEquals(Arrays.asList(TLS_V_1_3_VALUE), sslConfiguration.getEnabledProtocols());
+        Assert.assertEquals(Arrays.asList(TEST_CIPHER_SUITE_VALUE), sslConfiguration.getEnabledCipherSuites());
+    }
+
+    @Test(expected = NullPointerException.class)
```

Comment:
The latest version of JUnit 4 has deprecated the `expected` annotation property for exception checking in favor of using `assertThrows()`.  Since this is a new test class, it would be helpful to follow the recommended approach.  Using `assertThrows()` also provides the ability to have a more granular way to confirm that a particular method is throwing the exception.